Code/Mantid/Framework/Kernel/src/InternetHelper.cpp

```diff
@@ -403,5 +404,152 @@ int InternetHelper::downloadFile(const std::string &urlFile,
 **/
 void InternetHelper::setTimeout(int seconds) { m_timeout = seconds; }
 
+
+/** Gets the timeout in seconds
+* @returns The value in seconds for the timeout
+**/
+int InternetHelper::getTimeout() { return m_timeout; }
+
+/** Sets the Method
+* @param method A string of GET or POST, anything other than POST is considered GET
+**/
+void InternetHelper::setMethod(const std::string& method) { 
+  if (method == "POST") {
+    m_method = method; 
+  } else {
+    m_method = "GET"; 
+  }
+}
+
+/** Gets the method
+* @returns either "GET" or "POST"
+**/
+const std::string& InternetHelper::getMethod() {return m_method; }
+
+/** Sets the Content Type
+* @param contentType A string of the content type
+**/
+void InternetHelper::setContentType(const std::string& contentType) { 
+  m_contentType = contentType; 
+}
+
+/** Gets the Content Type
+* @returns A string of the content type
+**/
+const std::string& InternetHelper::getContentType() {return m_contentType; }
+
+
+
+/** Sets the content length
+* @param length The content length in bytes
+**/
+void InternetHelper::setContentLength(std::streamsize length) { m_contentLength = length; }
+
+/** Gets the content length
+* @returns The content length in bytes
+**/
+std::streamsize InternetHelper::getContentLength() { return m_contentLength; }
+
+/** Sets the body & content length  for future requests, this will also 
+*   set the method to POST is the body is not empty
+*   and GET if it is.
+* @param body A string of the body
+**/
+void InternetHelper::setBody(const std::string& body) { 
+  //clear the old body
+  m_body.str("") ;
+  if (body.empty()) {
+    setMethod("GET");
+  } else {
+    setMethod("POST");
+    m_body << body;
+  }
+  setContentLength(body.size());
+}
+
+/** Sets the body & content length  for future requests, this will also 
+*   set the method to POST is the body is not empty
+*   and GET if it is.
+* @param body A stringstream of the body
+**/
+void InternetHelper::setBody(const std::ostringstream& body) { 
+    setBody(body.str());
+}
+
+/** Sets the body & content length for future requests, this will also 
+*   set the method to POST is the body is not empty
+*   and GET if it is.
+* @param body A HTMLform
+**/
+void InternetHelper::setBody(Poco::Net::HTMLForm& form) { 
```

Comment:
The name in the doxygen tag "body" != "form", name of the parameter. I think this will generate a doxygen warning.
Could you change the doxygen tag to match the name of the parameter?